### PR TITLE
add OapRuntime for easy control

### PR DESCRIPTION
**Pull request by adrian-wang**
## What changes were proposed in this pull request?

This PR adds an `OapRuntime` class to manage some oap related objects, like `FiberCacheManager`, `MemoryManager` etc.,

## How was this patch tested?

unit test.

### CODE REVIEW

File: src/main/scala/org/apache/spark/SparkEnv.scala [hunk 1]
```diff
@@ -88,14 +84,14 @@ class SparkEnv (
     if (!isStopped) {
       isStopped = true
       pythonWorkers.values.foreach(_.stop())
+      OapRuntime.stop()
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/SparkEnv.scala]: For reviewers: this is the only change left to `SparkEnv.scala`

**Comment by linhong-intel** [comment]
So, with `OapRuntime.getOrCreate()` we can cover all spark entrance, right?

**Comment by adrian-wang** [comment]
@linhong-intel that is one possible solution. Another thinking is to insert explicit init() code in `SparkEnv` creation.

**Comment by linhong-intel** [comment]
@adrian-wang 
Understand your point. I like your change BTW.

File: src/main/scala/org/apache/spark/sql/oap/OapRuntime.scala [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      sparkEnv.blockManager,
+      fiberCacheManager,
+      sparkEnv.conf)
+  }
+  oapMetricsManager = new OapMetricsManager
+
+  override def stop(): Unit = {
+    if (OapRuntime.isLocal(sparkEnv.conf) && memoryManager != null) {
+      memoryManager.stop()
+      memoryManager = null
```
Comment (gczsjdy) [inline src/main/scala/org/apache/spark/sql/oap/OapRuntime.scala]: Do we have to set null after stop? I see in Spark we don't have to do this

**Comment by gczsjdy** [comment]
@adrian-wang I also like your code! Now only `StatisticsManager` as object left, but it seems to be a utils class, so don't need to be in `RunTime`

**Comment by LuciferYang** [comment]
Is it necessary for us to ensure synchronization of OapRuntime init and stop method ?

**Comment by linhong-intel** [comment]
LGTM, thanks @adrian-wang

File: src/main/scala/org/apache/spark/sql/oap/OapRuntime.scala (hunk 2 above)
Comment (adrian-wang) [reply src/main/scala/org/apache/spark/sql/oap/OapRuntime.scala]: when we call `stop()` from `OapRuntime.stop()`, we always unset the reference to the `OapRuntime` instance, so it is OK to not set to `null` here.

**Pull request merged**